# 1. Disable pointer compression (limits the maximum number of WasmVMs).
# 2. Don't override v8_target_cpu (https://crrev.com/c/3445204).

diff --git a/BUILD.bazel b/BUILD.bazel
index 1cc0121e60..4947c1dba2 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -161,7 +161,7 @@ v8_int(
 # If no explicit value for v8_enable_pointer_compression, we set it to 'none'.
 v8_string(
     name = "v8_enable_pointer_compression",
-    default = "none",
+    default = "False",
 )
 
 # Default setting for v8_enable_pointer_compression.
diff --git a/bazel/defs.bzl b/bazel/defs.bzl
index fe27db305b..dee5e69cc4 100644
--- a/bazel/defs.bzl
+++ b/bazel/defs.bzl
@@ -372,6 +372,12 @@ def v8_torque(name, noicu_srcs, icu_srcs, args, extras):
     )
 
 def _v8_target_cpu_transition_impl(settings, attr):
+    # Check for an existing v8_target_cpu flag.
+    if "@v8//bazel/config:v8_target_cpu" in settings:
+        if settings["@v8//bazel/config:v8_target_cpu"] != "none":
+            return
+
+    # Auto-detect target architecture based on the --cpu flag.
     mapping = {
         "haswell": "x64",
         "k8": "x64",
@@ -395,7 +401,7 @@ def _v8_target_cpu_transition_impl(settings, attr):
 # on the command line.
 v8_target_cpu_transition = transition(
     implementation = _v8_target_cpu_transition_impl,
-    inputs = ["//command_line_option:cpu"],
+    inputs = ["@v8//bazel/config:v8_target_cpu", "//command_line_option:cpu"],
     outputs = ["@v8//bazel/config:v8_target_cpu"],
 )
 
